internal/generate/olm-catalog/package_manifest_test.go

```diff
@@ -61,3 +122,95 @@ const packageManifestExp = `channels:
 defaultChannel: stable
 packageName: memcached-operator
 `
+
+func Test_NewPackageManifest(t *testing.T) {
+	type args struct {
+		operatorName string
+		channelName  string
+		version      string
+	}
+	tests := []struct {
+		name string
+		args args
+		want registry.PackageManifest
+	}{
+		{
+			name: "Should return a valid registry.PackageManifest",
+			want: registry.PackageManifest{
+				PackageName: "memcached-operator",
+				Channels: []registry.PackageChannel{
+					registry.PackageChannel{
+						Name:           "stable",
+						CurrentCSVName: "memcached-operator.v0.0.3",
+					},
+				},
+				DefaultChannelName: "stable",
+			},
+			args: args{
+				operatorName: testProjectName,
+				channelName:  "stable",
+				version:      csvVersion,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := newPackageManifest(tt.args.operatorName, tt.args.channelName, tt.args.version); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("newPackageManifest() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_Generate(t *testing.T) {
```

Comment:
@estroz check that here, for example, we have no args. 
Instead of that, we have structure fields which will mock the data.

Reply:
I recommend constructing a `Generator` test like this [CRD test](https://github.com/operator-framework/operator-sdk/blob/152e5d5/internal/generate/crd/crd_test.go#L44). This way, files can be generated on disk and cleaned up afterwards.

Can you please move all tests not related to validation to a follow-up PR? Thanks.